ddl/partition.go

```diff
@@ -660,9 +660,11 @@ func buildRangePartitionDefinitions(ctx sessionctx.Context, defs []*ast.Partitio
 
 func checkPartitionValuesIsInt(ctx sessionctx.Context, def *ast.PartitionDefinition, exprs []ast.ExprNode, tbInfo *model.TableInfo) error {
 	tp := types.NewFieldType(mysql.TypeLonglong)
-	if isColUnsigned(tbInfo.Columns, tbInfo.Partition) {
+	col := model.FindColumnInfo(tbInfo.Columns, tbInfo.Partition.Expr[1:len(tbInfo.Partition.Expr)-1])
```

Comment:
`tbInfo.Partition.Expr` is a string with single quotes，so `tbInfo.Partition.Expr[1:len(tbInfo.Partition.Expr)-1]`.